network.cpp

```diff
@@ -105,6 +105,26 @@ std::vector<weight> register_n_mutations(network n, double mut_rate, double mut_
     return  networks_weights;
 }
 
+std::vector<weight> register_n_activation_mutations(network n, double mut_rate, std::mt19937 &rng, int repeats)
+{
+    std::vector<weight> networks_weights;
+    for(int i = 0; i != repeats; i++)
+    {   auto n_new = n;
+        n_new.mutate_activation(mut_rate, rng);
+        auto weights = n_new.get_net_weights();
+
+        for(size_t j=0; j != weights.size(); ++j)
+            for(size_t k=0; k != weights[j].size(); ++k)
+                for(size_t l=0; l != weights[j][k].size(); ++l)
+                {
+                    if(!weights[j][k][l].is_active())
+                        networks_weights.push_back(weights[j][k][l]);
```

Comment:
I can do that for sure but I am not sure I understand your explanations. It will make sense with the other changes though, but maybe we can discuss the reasoning at some point?